### PR TITLE
(PUP-1952) Get tests working with jruby

### PR DESCRIPTION
**Pull request by pcarlisle**
Make the rspec tests run with JRuby 9.2.

### CODE REVIEW

**Comment by joshcooper** [comment]
I'm find changing to an external json fact, because we have an acceptance test for external executable facts: https://github.com/puppetlabs/puppet/blob/master/acceptance/tests/face/4654_facts_face.rb#L19-L27

File: lib/puppet/pops/puppet_stack.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ module Puppet::Pops
 #
 module PuppetStack
   # Pattern matching an entry in the ruby stack that is a puppet entry
-  PP_ENTRY_PATTERN = /^(.*\.pp)?:([0-9]+):in (`stack'|`block in call_function')/
+  PP_ENTRY_PATTERN = /^(.*\.pp)?:([0-9]+):in (`stack'|`block in call_function'|`<eval>')/
```
Comment (joshcooper) [inline lib/puppet/pops/puppet_stack.rb]: Why `<eval>`?

**Comment by joshcooper** [comment]
jenkins please test this on redhat7-64a,windows2012r2-64a

File: lib/puppet/pops/puppet_stack.rb (hunk 1 above)
Comment (pcarlisle) [reply lib/puppet/pops/puppet_stack.rb]: I don't know

**Pull request merged**